src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -1487,6 +1486,8 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
       compileUnitsInternal(units, fromPhase)
 
     private def compileUnitsInternal(units: List[CompilationUnit], fromPhase: Phase) {
+      def currentTime = java.util.concurrent.TimeUnit.NANOSECONDS.toMillis(System.nanoTime())
+
```

Comment:
This change is consistent with other places where the change from `System.currentTimeMillis` to `System.nanoTime` took place, when the issue was introduced. I don't mind using nanos consistently, but that seemed outside the scope of fixing the bug :)

Also, notice that startTime is defined in two scopes here, one for the entire process and once per phase. That's why I opted for the method.